### PR TITLE
ci: implement initial ci on github actions

### PR DESCRIPTION
**Pull request by PastaPastaPasta**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+jobs:
+  build_and_test:
+    name: Rust project
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+      - run: cargo build
```
Comment (shumkov) [inline .github/workflows/ci.yml:14]: Shouldn't it be cargo test?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      - run: cargo test
```
Comment (shumkov) [inline .github/workflows/ci.yml:15]: Let's do it as two separate jobs to run them in parallel.

Comment (shumkov) [reply .github/workflows/ci.yml:15]: I would also add one more job with `cargo clippy -- -D warnings` to check formatting

**Review by shumkov — CHANGES_REQUESTED** [review]

**Comment by fominok** [comment]
Is is possible to run formatter automatically?

**Comment by shumkov** [comment]
> Is is possible to run formatter automatically?

Not sure we want CI to commit code

File: .github/workflows/ci.yml (hunk 1 above)
Comment (PastaPastaPasta) [reply .github/workflows/ci.yml:14]: The goal is to have it in two steps so you can easily identify whether just test failed, or just build failed

File: .github/workflows/ci.yml (hunk 2 above)
Comment (PastaPastaPasta) [reply .github/workflows/ci.yml:14]: done

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  clippy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+      - run: rustup component add clippy --toolchain nightly
+      - run: cargo clippy
```
Comment (PastaPastaPasta) [inline .github/workflows/ci.yml:23]:
```suggestion
  clippy:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions-rs/toolchain@v1
        with:
          toolchain: nightly
      - run: rustup component add clippy --toolchain nightly
      - run: cargo clippy
    cargo-check:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions-rs/toolchain@v1
        with:
          toolchain: nightly
      - run: cargo check
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (PastaPastaPasta) [reply .github/workflows/ci.yml:13]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+on: [push, pull_request]
+
+name: CI
+
+jobs:
+  build_and_test:
```
Comment (shumkov) [inline .github/workflows/ci.yml:6]: Rename to `test`

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+      - run: cargo test
+  clippy:
```
Comment (shumkov) [inline .github/workflows/ci.yml:14]: Let's rename to `linting`

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+      - run: rustup component add clippy --toolchain nightly
+      - run: cargo clippy
+  cargo-check:
```
Comment (shumkov) [inline .github/workflows/ci.yml:23]: Let's rename to `errors`

**Review by shumkov — CHANGES_REQUESTED** [review]
We need to fix all warning before can merge this PR

File: .github/workflows/ci.yml (hunk 4 above)
Comment (PastaPastaPasta) [reply .github/workflows/ci.yml:6]:
```suggestion
  test:
```

File: .github/workflows/ci.yml (hunk 5 above)
Comment (PastaPastaPasta) [reply .github/workflows/ci.yml:14]:
```suggestion
  lint:
```

File: .github/workflows/ci.yml (hunk 6 above)
Comment (PastaPastaPasta) [reply .github/workflows/ci.yml:23]:
```suggestion
  errors:
```

**Comment by PastaPastaPasta** [comment]
> We need to fix all warning before can merge this PR

No, if that's the case, then let's remove the linting for now, so that we can at least ensure we're not merging stuff that breaks compilation. We can fix the linter warnings later.

**Review by shumkov — APPROVED** [review]
LGTM!

**Pull request merged**